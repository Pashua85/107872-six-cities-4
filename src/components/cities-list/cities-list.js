import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {changeCityAction, getPlacesAction} from '../../action-creators/action-creators';

const CitiesList = (props) => {
  const {cityNames, city, onClick} = props;
  const list = cityNames.map((cityName) => {
    if (cityName === city) {
      return (
        <li className="locations__item" key={cityName}>
          <a className="locations__item-link tabs__item tabs__item--active">
            <span>{cityName}</span>
          </a>
        </li>
      );
    } else {
      return (
        <li className="locations__item" key={cityName}>
          <a className="locations__item-link tabs__item" href="#" onClick={() => onClick(cityName)}>
            <span>{cityName}</span>
          </a>
        </li>
      );
    }
  });
  return (
    <section className="locations container">
      <ul className="locations__list tabs__list">
        {list}
      </ul>
    </section>
  );
};

CitiesList.propTypes = {
  city: PropTypes.string.isRequired,
  cityNames: PropTypes.arrayOf(PropTypes.string).isRequired,
  onClick: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  city: state.city,
  cityNames: state.cityNames
});

const mapDispatchToProps = (dispatch) => ({
  onClick: (city) => {
    dispatch(changeCityAction(city));
    dispatch(getPlacesAction(city));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(CitiesList);
export {CitiesList};

