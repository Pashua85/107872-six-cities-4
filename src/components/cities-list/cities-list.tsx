import React, { Dispatch } from 'react';
import {connect} from 'react-redux';
import ActionCreator from '../../store/action-creator/action-creator';
import {getCities} from '../../store/reducers/cities-reducer/selectors';
import {getCurrentCity} from '../../store/reducers/currentCityReducer/selectors';
import {ICity, IStore} from '../../types/types';

interface CitiesListProps {
  cities: ICity[],
  currentCity: ICity,
  onClick: (city: ICity) => void
}

const CitiesList: React.FC<CitiesListProps> = (props) => {
  const {cities, currentCity, onClick} = props;

  const list = cities.map((city) => {
    if (city.name === currentCity.name) {
      return (
        <li className="locations__item" key={city.name}>
          <a className="locations__item-link tabs__item tabs__item--active">
            <span>{city.name}</span>
          </a>
        </li>
      );
    } else {
      return (
        <li className="locations__item" key={city.name}>
          <a className="locations__item-link tabs__item" href="#" onClick={() => onClick(city)}>
            <span>{city.name}</span>
          </a>
        </li>
      );
    }
  });
  return (
    <section className="locations container">
      <ul className="locations__list tabs__list">
        {list}
      </ul>
    </section>
  );
};

const mapStateToProps = (state: IStore) => ({
  currentCity: getCurrentCity(state),
  cities: getCities(state)
});

const mapDispatchToProps = (dispatch: Dispatch<any>) => ({
  onClick: (city: ICity) => {
    dispatch(ActionCreator.setCurrentCity(city));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(CitiesList);
export {CitiesList};

