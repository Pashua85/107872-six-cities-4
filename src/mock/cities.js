import {v4 as uuidv4} from 'uuid';

export default [
  {
    id: uuidv4(),
    cityName: `Paris`,
    places: []
  },
  {
    id: uuidv4(),
    cityName: `Cologne`,
    places: []
  },
  {
    id: uuidv4(),
    cityName: `Brussels`,
    places: []
  },
  {
    id: uuidv4(),
    cityName: `Amsterdam`,
    places: [
      {
        id: uuidv4(),
        propertyName: `Beautiful luxurious apartment at great location`,
        propertyType: `apartment`,
        propertyText: [
          `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
          `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
        ],
        price: 120,
        rating: 3.9,
        isPremium: true,
        bedroomsAmount: 3,
        guestMax: 4,
        propertyItems: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
        host: {
          name: `Angelina`,
          avatar: `avatar-angelina.jpg`,
          isSuper: true
        },
        titlePhoto: `apartment-01.jpg`,
        photos: [
          {
            title: `room.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-01.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-02.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-03.jpg`,
            id: uuidv4()
          },
          {
            title: `studio-01.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-01.jpg`,
            id: uuidv4()
          }
        ],
        reviews: [
          {
            id: uuidv4(),
            userName: `Max`,
            avatar: `img/avatar-max.jpg`,
            rating: 4,
            text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
          }
        ],
        coords: [52.3909553943508, 4.85309666406198]
      },
      {
        id: uuidv4(),
        propertyName: `Wood and stone place`,
        propertyType: `room`,
        propertyText: [
          `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
          `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
        ],
        price: 80,
        rating: 4.6,
        isPremium: false,
        bedroomsAmount: 1,
        guestMax: 2,
        propertyItems: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
        host: {
          name: `Angelina`,
          avatar: `avatar-angelina.jpg`,
          isSuper: true
        },
        titlePhoto: `room.jpg`,
        photos: [
          {
            title: `room.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-01.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-02.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-03.jpg`,
            id: uuidv4()
          },
          {
            title: `studio-01.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-01.jpg`,
            id: uuidv4()
          }
        ],
        reviews: [],
        coords: [52.369553943508, 4.85309666406198]
      },
      {
        id: uuidv4(),
        propertyName: `Canal View Prinsengracht`,
        propertyType: `apartment`,
        propertyText: [
          `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
          `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
        ],
        price: 132,
        rating: 4.7,
        isPremium: false,
        bedroomsAmount: 3,
        guestMax: 4,
        propertyItems: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
        host: {
          name: `Angelina`,
          avatar: `avatar-angelina.jpg`,
          isSuper: true
        },
        titlePhoto: `apartment-02.jpg`,
        photos: [
          {
            title: `room.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-01.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-02.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-03.jpg`,
            id: uuidv4()
          },
          {
            title: `studio-01.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-01.jpg`,
            id: uuidv4()
          }
        ],
        reviews: [],
        coords: [52.3909553943508, 4.929309666406198]
      },
      {
        id: uuidv4(),
        propertyName: `Nice, cozy, warm big bed apartment`,
        propertyType: `apartment`,
        propertyText: [
          `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
          `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
        ],
        price: 180,
        rating: 5,
        isPremium: true,
        bedroomsAmount: 4,
        guestMax: 5,
        propertyItems: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
        host: {
          name: `Angelina`,
          avatar: `avatar-angelina.jpg`,
          isSuper: true
        },
        titlePhoto: `apartment-03.jpg`,
        photos: [
          {
            title: `room.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-01.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-02.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-03.jpg`,
            id: uuidv4()
          },
          {
            title: `studio-01.jpg`,
            id: uuidv4()
          },
          {
            title: `apartment-01.jpg`,
            id: uuidv4()
          }
        ],
        reviews: [],
        coords: [52.3809553943508, 4.939309666406198]
      }
    ]
  },
  {
    id: uuidv4(),
    cityName: `Hamburg`,
    places: []
  },
  {
    id: uuidv4(),
    cityName: `Dusseldorf`,
    places: []
  }
];
