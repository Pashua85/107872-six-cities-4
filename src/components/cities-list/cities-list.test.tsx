import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {CitiesList} from './cities-list';
import cities from '../../mock/test-cities';


Enzyme.configure({
  adapter: new Adapter()
});

describe(`CitesList`, () => {
  it(`should render correctly`, () => {
    const wrapper = shallow(
        <CitiesList currentCity={cities[0]} cities={cities} onClick={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
