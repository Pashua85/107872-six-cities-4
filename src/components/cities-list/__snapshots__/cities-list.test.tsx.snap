// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitesList should render correctly 1`] = `ShallowWrapper {}`;
